internal/pgdb/pgdb_field.go

```diff
@@ -389,46 +389,10 @@ func getCommonFields(ctx pgsgo.Context, m pgs.Message, ix *importTracker) ([]*fi
 		if ext.MessageBehavoir != pgdb_v1.FieldOptions_MESSAGE_BEHAVOIR_VECTOR {
 			continue
 		}
-		if !field.Type().IsRepeated() {
-			panic(fmt.Errorf("pgdb: vector behavior only supported on repeated fields: %s", field.FullyQualifiedName()))
-		}
-		subMsg := field.Type().Element().Embed()
-		if subMsg == nil {
-			panic(fmt.Errorf("pgdb: vector behavior only supported on message fields: %s", field.FullyQualifiedName()))
-		}
-		allFields := subMsg.Fields()
-		if len(allFields) != 2 {
-			panic(fmt.Errorf("pgdb: vector message must only have model enum and float array: %s", field.FullyQualifiedName()))
-		}
 
-		var enumField pgs.Field
-		var floatField pgs.Field
-		var vectorSize int32
-
-		for _, subField := range allFields {
-			switch subField.Descriptor().GetNumber() {
-			case 1:
-				// enum
-				if subField.Type().ProtoType() != pgs.EnumT {
-					panic(fmt.Errorf("pgdb: vector message must have model enum as first field: %s", field.FullyQualifiedName()))
-				}
-				enumField = subField
-			case 2:
-				// repeated float
-				if !subField.Type().IsRepeated() || subField.Type().Element().ProtoType() != pgs.FloatT {
-					panic(fmt.Errorf("pgdb: vector message must have repeated float as second field: %s", field.FullyQualifiedName()))
-				}
-				subExt := pgdb_v1.FieldOptions{}
-				_, err := subField.Extension(pgdb_v1.E_Options, &subExt)
-				if err != nil {
-					return nil, fmt.Errorf("pgdb: getField: failed to extract Message extension from '%s': %w", field.FullyQualifiedName(), err)
-				}
-				if subExt.VectorSize == 0 {
-					panic(fmt.Errorf("pgdb: vector message must have vector_size set on repeated float field: %s", field.FullyQualifiedName()))
-				}
-				vectorSize = subExt.VectorSize
-				floatField = subField
-			}
+		enumField, floatField, vectorSize, err := pgdb_v1.GetFieldVectorShape(field)
```

Comment:
this should be in the internal package, not in the pgpdb/v1 package -- that package is a runtime package, internal is code generation time

Reply:
ok I originally put it there then changed my mind 😓

Reply:
https://github.com/ductone/protoc-gen-pgdb/pull/66/commits/ee4ecbcba6b2f2d3c6f08f594fa411498d7dd6d5